settlement/paypal/data.go

```diff
@@ -0,0 +1,174 @@
+package paypal
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"time"
+
+	"github.com/brave-intl/bat-go/utils/altcurrency"
+	uuid "github.com/satori/go.uuid"
+	"github.com/shengdoushi/base58"
+	"github.com/shopspring/decimal"
+)
+
+var (
+	supportedCurrencies = map[string]float64{
+		"JPY": 0,
+		"USD": 2,
+	}
+)
+
+// Args defines all inputs from cli
+type Args struct {
+	In       string
+	Currency string
+	Auth     string
+	Date     string
+	Rate     decimal.Decimal
+	Out      string
+}
+
+// CheckDate parses a date from the input string to validate it
+func (args Args) CheckDate() (time.Time, error) {
+	return time.Parse(time.RFC3339, args.Date+"T00:00:00.0Z")
+}
+
+// CurrencyPrices is a hash of all currencies supported
+type CurrencyPrices struct {
+	JPY decimal.Decimal `json:"JPY"`
+}
+
+// RateResponse is the response received from ratios
+type RateResponse struct {
+	LastUpdated time.Time                  `json:"lastUpdated"`
+	Payload     map[string]decimal.Decimal `json:"payload"`
+}
+
+// RefIDKey is used to generate a hash
+func (pm *Metadata) RefIDKey(channel string) string {
+	return pm.Prefix + pm.Publisher + channel
+}
+
+// GenerateRefID converts a hex to base62
+func (pm *Metadata) GenerateRefID(channel string) string {
+	key := pm.RefIDKey(channel) // nothing more needed for now
+	bytes := sha256.Sum256([]byte(key))
+	refID := base58.Encode(bytes[:], base58.BitcoinAlphabet)
+	refID = refID[:30]
+	pm.RefID = refID
+	return refID
+}
+
+// Note a note to be transformed into personal email message
+type Note struct {
+	Channel string
+	Probi   decimal.Decimal
+}
+
+// Metadata holds metadata to create a row for paypal
+type Metadata struct {
+	ExecutedAt    time.Time
+	Rate          decimal.Decimal
+	Prefix        string
+	Section       string
+	PayerID       string
+	Channel       string
+	Publisher     string
+	Probi         decimal.Decimal
+	Currency      string
+	RefID         string
+	Note          []Note
+	NoteDelimiter string
+}
+
+// EyeshadeTransaction an object to express the flow of funds
+type EyeshadeTransaction struct {
+	Publisher     string                  `json:"publisher"`
+	Owner         string                  `json:"owner"`
+	Probi         decimal.Decimal         `json:"probi"`
+	Currency      string                  `json:"currency"`
+	Hash          string                  `json:"hash"`
+	Amount        decimal.Decimal         `json:"amount"`
+	ExecutedAt    time.Time               `json:"executedAt"`
+	AltCurrency   altcurrency.AltCurrency `json:"altcurrency"`
+	Address       uuid.UUID               `json:"address"`
+	Fees          decimal.Decimal         `json:"fees"`
+	Fee           decimal.Decimal         `json:"fee"`
+	Commission    decimal.Decimal         `json:"commission"`
+	TransactionID uuid.UUID               `json:"transactionId"`
+	Type          string                  `json:"type"`
+}
+
+// NewMetadata creates a new paypal metadata object
+func NewMetadata(pm Metadata) Metadata {
```

Comment:
this function seems very strange to me, we're just using this to fill defaults in a couple fields?